docs/content/useOnOutsideClick.mdx

```diff
@@ -0,0 +1,40 @@
+---
+title: useOnOutsideClick
+---
+
+`useOnOutsideClick` is a utility Hook that calls a user provided callback function when the user clicks outside of the provided container.
+
+You can also pass an array of `ignoredRefs` to prevent calling the callback function on additional elements on the page. This can be handy for ignoring clicks on trigger buttons that already manage the open/closed state of content.
+
+
+### Usage
+
+```jsx live
+  <State>
+    {([isOpen, setIsOpen]) => {
+      const containerRef = React.useRef(null)
+      const triggerRef = React.useRef(null)
+      useOnOutsideClick({onClickOutside: () => setIsOpen(false), containerRef, ignoreClickRefs: [triggerRef]})
```

Comment:
Instead of defining an inline function for `onClickOutside`, can we properly wrap it in `useCallback`? Since this is a code sample it has a high probability of being copied, so we want to make sure folks do the right thing.

Alternatively, `useOnOutsideClick` can take an additional set of dependencies (similar to `useAnchoredPosition`) and add those to _its_ `useCallback` dependency list instead of `onClickOutside`. It can be hard to think about how multiple levels of `useCallback` interact - let me know if you want me to clarify this suggestion.

Reply:
done! can you check to confirm this is what you were thinking?

Reply:
I'm not sure what has changed. Did you push the fix for this?

Reply:
Never mind; the diff was showing me the old version :doh: